src/site/content/en/blog/disallow-synchronous-xhr/index.md

```diff
@@ -0,0 +1,109 @@
+a bad user experience by delaying closing of the page for up to several seconds.
+
+This practice needs to change, and browsers are responding. The `XMLHttpRequest()`
+specification is already [slated for deprecation and
+removal](https://xhr.spec.whatwg.org/#sync-warning). Chrome 80 takes the first
+step by disallowing synchronous calls inside several event handlers,
+specifically `beforeunload`, `unload`, `pagehide`, and `visibilitychange` when
+they are fired in the dismissal. WebKit also recently landed [a commit implementing
+the same behavior change](https://bugs.webkit.org/show_bug.cgi?id=204912).
+
+In this article I'll briefly describe options for those who need time to update
+their sites and outline the alternatives to `XMLHttpRequest()`.
+
+## Temporary opt-outs
+
+Chrome does not simply want to pull the plug on `XMLHttpRequest()`, which is why a few
+temporary opt-out options are available. For sites on the internet, [an origin
+trial is
+available](https://developers.chrome.com/origintrials/#/view_trial/4391009636686233601).
+With this, you add an origin-specific token to your page headers that enables
+synchronous `XMLHttpRequest()` calls. This option ends shortly before Chrome 86
+ships, sometime in late October of 2020. Enterprise Chrome customers can also
+use the  `AllowSyncXHRInPageDismissal` policy flag, which ends at the same time.
+
+## Alternatives
+
+Regardless of how you send data back to the server, it's best to avoid waiting
+until page unload to send all the data at once. Aside from creating a bad user
+experience, you risk data loss if something goes wrong.  Unload events [often
+don't fire on mobile
+browsers](https://www.igvita.com/2015/11/20/dont-lose-user-and-app-state-use-page-visibility/)
+because there are [many ways to
+close](https://developers.google.com/web/updates/2018/07/page-lifecycle-api) a
+tab or browser on mobile operating systems without the unload event firing. With
+`XMLHttpRequest()`, using small payloads was a choice. Now it's a requirement. Both of
+its alternatives have an upload limit of 64&nbsp;KB per context, as required
+by the specification.
+
+### Fetch keepalive
+
+The [Fetch API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API)
+provides a robust means of dealing with server interactions and [a consistent
+interface](https://fetch.spec.whatwg.org/#preface) for use across different
+platform APIs. Among its options is one that ensures a request continues whether
+or not the page that made it stays open. This is what `keepalive` does, as I've
+shown below:
+
+```js/5
+window.addEventListener('unload', {
+  fetch('/siteAnalytics', {
+    method: 'POST',
+    body: getStatistics(),
+    keepalive: true
+  });
+}
+```
+
+The `fetch()` method has the advantage of greater control over what's sent to
+the server. What I don't show in the example is that `fetch()` also returns a
+promise that resolves with a Response object. Since I'm trying to get out of the
```

Comment:
```suggestion
promise that resolves with a `Response` object. Since I'm trying to get out of the
```

Because we're referring to a specific API I think it also makes sense to format this as inline code